packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -484,23 +557,30 @@ class IconButton extends StatelessWidget {
         adjustedStyle = style!.merge(adjustedStyle);
       }
 
+      Widget adjustedIcon = icon;
+      if (isSelected != null && isSelected! && selectedIcon != null) {
```

Comment:
```suggestion
      if (isSelected == true && selectedIcon != null) {
```

Reply:
Thanks! Changed to `(isSelected ?? false)` based on the use_if_null_to_convert_nulls_to_bools lint.

Reply:
Nice that is even better.